core/src/main/scala/org/apache/spark/executor/ExecutorMetricsSource.scala

```diff
@@ -19,9 +19,18 @@ package org.apache.spark.executor
 
 import com.codahale.metrics.{Gauge, MetricRegistry}
 
-import org.apache.spark.metrics.ExecutorMetricType
+import org.apache.spark.metrics.{ExecutorMetricType, MetricsSystem}
 import org.apache.spark.metrics.source.Source
 
+// Expose executor metrics from [[ExecutorMetricsType]] using the Dropwizard metrics system
+// Metrics related the memory system can be expensive to gather, we implement some optimizations:
+// (1) Metrics values are cached, updated at each heartbeat (default period is 10 seconds)
+// An alternative faster polling mechanism is used only if activated, by setting
+// spark.executor.metrics.pollingInterval=<interval in ms>
+// (2) procfs metrics are gathered all in one-go and only conditionally:
+// if the /proc filesystem exists
+// and spark.eventLog.logStageExecutorProcessTreeMetrics.enabled=true
+// and spark.eventLog.logStageExecutorMetrics.enabled=true
```

Comment:
can you change to javadoc style? (`/**`)  just helps in IDEs

also some nits:
blank line after the first summary line, end sentence with period.

missing "to" and "so" in "Metrics related to the memory system can be expensive to gather, so ..."